src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,119 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Storage class deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Constructor for Storage.
+     *
+     * @param filePath the string that represents the path of the text file to
+     *                 load tasks from or save tasks in
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Returns an ArrayList of Task Objects that is loaded from the text file.
+     * Returns an empty ArrayList if the text file is empty or does not exist.
+     *
+     * @return an ArrayList of Task Objects
+     * @throws DukeException if there is an error in accessing the file
+     */
+    public ArrayList<Task> loadData() throws DukeException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        if (createDukeTextFile()) {
+            try {
+                File dukeFile = new File(filePath);
+                Scanner myReader = new Scanner(dukeFile);
+                while (myReader.hasNextLine()) {
+                    String data = myReader.nextLine();
+                    char taskType = data.charAt(0);
+                    switch (taskType) {
+                    case 'T':
+                        String[] todoSplit = data.split(" // ", 3);
```

Comment:
Perhaps you can consider putting the logic for handling these cases in separate methods and simply call the methods for these different cases? Since the code for these 3 cases are somewhat similar, it may make your code less verbose and even simpler 😄